pandas/core/dtypes/dtypes.py

```diff
@@ -220,10 +218,10 @@ def __eq__(self, other):
             # CDT(., .) = CDT(None, False) and *all*
             # CDT(., .) = CDT(None, True).
             return True
-        elif self.ordered:
-            return other.ordered and self.categories.equals(other.categories)
-        elif other.ordered:
-            return False
+        elif self.ordered or other.ordered:
+            # at least one ordered
```

Comment:
I would expand on the comment here (kind of like you did in the tests below), to enumerate the cases for ordered == ordered (and when they match / don't match)